### PR TITLE
Fix heap-buffer-overflow issues in wolfSSL_SMIME_read_PKCS7

### PR DESCRIPTION
**Pull request by embhorn**
This issue was first reported in ZD12233

### CODE REVIEW

**Review by kabuobeid — APPROVED** [review]

**Comment by embhorn** [comment]
Retest this please

File: src/ssl.c [hunk 1]
```diff
@@ -54583,16 +54583,25 @@ WOLFSSL_API PKCS7* wolfSSL_SMIME_read_PKCS7(WOLFSSL_BIO* in,
...
+            }
+
             /* Strip the final trailing newline.  Support \r, \n or \r\n. */
             if (canonSection[canonPos] == '\n') {
-                canonPos--;
-                if (canonSection[canonPos] == '\r') {
+                if (canonPos > 0) {
                     canonPos--;
                 }
+                if (canonSection[canonPos] == '\r') {
```
Comment (SparkiDev) [inline src/ssl.c:54595]: Don't need this as a separate case.
Remove this and instead of 'else if' replace with 'if'.

File: src/ssl.c [hunk 2]
```diff
@@ -54583,16 +54583,25 @@ WOLFSSL_API PKCS7* wolfSSL_SMIME_read_PKCS7(WOLFSSL_BIO* in,
                 }
             }
 
-            canonPos--;
+            if (canonPos > 0) {
```
Comment (SparkiDev) [inline src/ssl.c:54586]: include all the stuff below in this block rather than checking each time.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 2 above)
Comment (SparkiDev) [reply src/ssl.c:54586]: You are correct. Bad request!

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c:54595]: check

**Comment by SparkiDev** [comment]
retest this please

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**